modules/querying/pages/query-operations.adoc

```diff
@@ -1142,3 +1142,104 @@ The scope of the command depends on the user's current scope.
 If the user has set a working graph, then `DROP ALL` removes all the queries for that graph.
 If a user has set their scope to be global, then `DROP ALL` removes all queries across all graph spaces.
 
+== Query Descriptions
+
+A query description provides a human-readable explanation of a query or its parameters, offering more detailed information to help users understand the query's purpose and functionality.
+Here, you will find how to update, show, and drop descriptions of queries and query parameters.
+
+=== Update description
+After creating a query, you can add a description for the query.
+
+*Syntax*
+[source]
+----
+UPDATE DESCRIPTION OF <objectType> <objectName> [ON GRAPH <graphName>] <description>
+----
+
+Required privilege
+`WRITE_QUERY`
+
+To update a description, run the `UPDATE DESCRIPTION` command as shown below. You can update the description of a query or a query parameter. If you choose not to specify a graph in the command, the current scope will be used as the scope of the description.
+
+[source, gsql]
+----
+GSQL > USE GRAPH poc_graph
+GSQL > UPDATE DESCRIPTION OF QUERY query1 "List of all employees in the company"
+GSQL > UPDATE DESCRIPTION OF QUERY query2 ON GRAPH graph1 "Details of employees in a specific department"
+GSQL > UPDATE DESCRIPTION OF QUERY_PARAM query1.param1 "Department where the employee works"
+----
+
+[NOTE]
+====
+The term `UPDATE` here means adding a description to the target.
+If the target does not already have a description, it will be added.
+If the target already has a description, it will be replaced with the new one.
+====
+
+[CAUTION]
+====
+* When updating a query, use `CREATE OR REPLACE QUERY` instead of `CREATE QUERY` to retain the query description.
+* `CREATE QUERY` may unintentionally drop both the query and its description if it already exists, whereas `CREATE OR REPLACE QUERY` ensures the description is retained by replacing the query without dropping it.
+====
+
+=== Show description
+
+The `SHOW DESCRIPTION` command displays the description of a query or a query parameter.
+
+*Syntax*
+[source]
+----
+SHOW DESCRIPTION OF <objectType> <objectName> [ON GRAPH <graphName>]
+----
+
+Required privilege
+`READ_QUERY`
+
+To display a description, use the `SHOW DESCRIPTION` command as shown below. Similar to the update command, you can display the description of a query or a query parameter.
+
+[source, gsql]
+----
+GSQL > USE GRAPH poc_graph
+GSQL > SHOW DESCRIPTION OF QUERY query1
+GSQL > SHOW DESCRIPTION OF QUERY query1*
+GSQL > SHOW DESCRIPTION OF QUERY * ON GRAPH poc_graph
+GSQL > SHOW DESCRIPTION OF QUERY_PARAM query1.param1
+GSQL > SHOW DESCRIPTION OF QUERY_PARAM query2.*
+----
+
+* For the `SHOW DESCRIPTION` command, you can display the descriptions of individual queries or query parameters. Each query or parameter must be specified separately, and you cannot combine them in a single request.
```

Comment:
```suggestion
```
We already defined SHOW DESCRIPTION above.  
There is no need to say "must be specified separately" if we clearly define what you CAN do in one statement. Focus on ABILITIES, not INABILITIES.  Only mention inabilities if the feature gap is surprising.